### PR TITLE
Setup travis CI to test upstream applications

### LINKED ISSUE #28: Implement pre-merge CI hooks that checks upstream apps can integrate
**Is your feature request related to a problem? Please describe.**
We know we are planning to use this framework within [simorgh](https://github.com/bbc/simorgh) and in the future for a World Service renderer. Because of this, we should create a CI process that clones each of the upstream applications against their default branch and then runs `npm install` and `npm link` to each SPArtacus feature branch before running the tests in each application. 

**Describe the solution you'd like**
Using Travis have logic which does the following:
- clones simorgh
- npm install
- npm link (in spartacus)
- npm link ../spartacus (in simorgh)
- npm run test

**Describe alternatives you've considered**
Use Jenkins? 

NB: the solution needs to be easily extendible for multiple upstream application

### PR DESCRIPTION
**Pull request by pjlee11**
Resolves #28 

This is a precursor to integrating SPArtacus into simorgh.

Finishes the logic in the `.travis.yml` to build the current feature branches `/lib/` then clones and installs simorgh, `npm link`'s SPArtacus and simorgh before running the following commands:

- Unit tests: `npm run test`
- Storybook integration tests: `xvfb-run npm run test:e2e:storybook:ci`
- CI audit: `npm run audit:ci`
- Production build of local bundles: `npm run build`
- Production CI tests: `xvfb-run npm run test:prod:ci`
- Lighthouse tests: `xvfb-run npm run test:lighthouse:ci`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,25 +5,24 @@ jobs:
...
-        - ./cc-test-reporter after-build -t lcov --debug --exit-code $TRAVIS_TEST_RESULT
         - xvfb-run npm run test:e2e:storybook:ci
-    # - stage: test
-    #   install: npm ci
-    #   script:
-    #     - npm run audit:ci
-    #     - npm run build
-    #     - xvfb-run npm run test:prod:ci
-    #     - xvfb-run npm run test:lighthouse:ci
+        - npm run audit:ci
```
Comment (bcmn) [inline .travis.yml]: I'm not sure I like that these values are hard-coded -- could we utilise the makefile or similar?

File: .travis.yml [hunk 2]
```diff
@@ -5,25 +5,24 @@ jobs:
...
-        - ./cc-test-reporter before-build
+        # - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ../cc-test-reporter
+        # - chmod +x ../cc-test-reporter
+        # - ../cc-test-reporter before-build
       script:
-        - npm run build
+        - npm run build # ensure SPArtacus/lib/ is built
         - npm link
+        - cd ../
         - git clone https://github.com/bbc/simorgh.git
```
Comment (bcmn) [inline .travis.yml]: I get why we're doing this -- we want to detect breaking changes in Spartacus before we make them, but... this feels strange. 

I think my instinct says that checking out dependents to test for breaking changes is a little backwards -- I'd instead want the _dependency's_ (spartacus') test suite to be comprehensive enough to give me confidence I don't need to do this

File: .travis.yml [hunk 3]
```diff
@@ -5,25 +5,24 @@ jobs:
...
       before_script:
-        - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
-        - chmod +x ./cc-test-reporter
-        - ./cc-test-reporter before-build
+        # - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ../cc-test-reporter
+        # - chmod +x ../cc-test-reporter
+        # - ../cc-test-reporter before-build
       script:
-        - npm run build
+        - npm run build # ensure SPArtacus/lib/ is built
```
Comment (dr3) [inline .travis.yml]: Would be nice to make this generic, like having a script to do this for you per repo 
```
./upstreamtest.sh https://github.com/bbc/simorgh.git simorgh
```
Which runs clone, install, cd and then `make canIntegrate` which runs all the tests that application agrees to run

**Comment by pjlee11** [comment]
We should allow the upstream application to define it's our test commands for checking we haven't broken integration. Therefore this will use `make canIntegrate` which can then be defined independently in each upstream application EG: https://github.com/bbc/simorgh/pull/1372

**Pull request merged**